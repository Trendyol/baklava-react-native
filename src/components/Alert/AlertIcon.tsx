import React from 'react';
import Icon from '../Icon/Icon';
import { IconNameType } from '../Icon/types';
import { VariantProps } from '@ergenekonyigit/restyle';
import { Theme } from '../../theme';

export const AlertIcon = React.memo(
  ({
    variant = 'info',
    icon,
  }: {
    icon?: IconNameType | boolean | null;
    variant?: VariantProps<Theme, 'alertVariants'>['variant'];
  }): JSX.Element | null => {
    const iconName = React.useMemo(() => {
      if (typeof icon === 'boolean' && !icon) {
        return null;
      }

      return {
        info: 'info',
        success: 'check-fill',
        warning: 'warning',
        danger: 'alert',
      }[variant];
    }, [variant, icon]) as IconNameType | null;

    const iconColor = React.useMemo(() => {
      return {
        info: 'primaryKey',
        warning: 'warningKey',
        success: 'successKey',
        danger: 'dangerKey',
      }[variant];
    }, [variant]);

    if (!iconName) {
      return null;
    }

    return (
      <Icon size="s" name={iconName} color={iconColor} testID="alert-icon" />
    );
  },
);
