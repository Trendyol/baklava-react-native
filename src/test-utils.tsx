import { ThemeProvider } from '@ergenekonyigit/restyle';
import { render } from '@testing-library/react-native';
import * as React from 'react';
import {
  GestureResponderHandlers,
  PanResponder,
  PanResponderCallbacks,
} from 'react-native';
import theme from './theme';
import { PortalProvider } from '.';

theme.fonts = {
  light: 'Rubik-Light',
  regular: 'Rubik-Regular',
  medium: 'Rubik-Medium',
  semiBold: 'Rubik-SemiBold',
  bold: 'Rubik-Bold',
};

export interface RenderOptions {
  wrapper?: React.ComponentType<any>;
  createNodeMock?: (element: React.ReactElement<any>) => any;
}

const AllProviders = ({ children }: { children?: React.ReactNode }) => (
  <ThemeProvider theme={theme}>
    <PortalProvider>{children}</PortalProvider>
  </ThemeProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options }) as any;

const mockPanResponder = () => {
  jest
    .spyOn(PanResponder, 'create')
    .mockImplementation(
      ({
        onStartShouldSetPanResponderCapture,
        onMoveShouldSetPanResponder,
        onPanResponderGrant,
        onPanResponderMove,
        onPanResponderRelease,
      }: PanResponderCallbacks) => ({
        panHandlers: {
          onStartShouldSetResponderCapture: onStartShouldSetPanResponderCapture,
          onMoveShouldSetResponder: onMoveShouldSetPanResponder,
          onResponderGrant: onPanResponderGrant,
          onResponderMove: onPanResponderMove,
          onResponderRelease: onPanResponderRelease,
        } as GestureResponderHandlers,
      }),
    );
};

export * from '@testing-library/react-native';
export { customRender as render, mockPanResponder };
