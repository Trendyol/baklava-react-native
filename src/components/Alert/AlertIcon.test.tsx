import React from 'react';
import { render } from '../../test-utils';
import { AlertIcon } from './AlertIcon';

describe('AlertIcon', () => {
  test('should render alert icon correctly', () => {
    // when
    const { toJSON } = render(<AlertIcon />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render null when icon false', () => {
    // when
    const { toJSON } = render(<AlertIcon icon={false} />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render alert icon with given variant', () => {
    // when
    const { getByTestId } = render(<AlertIcon variant="success" />);

    const alertComponent = getByTestId('alert-icon');

    // then
    expect(alertComponent.props.title).toBe('check-fill');
  });
});
