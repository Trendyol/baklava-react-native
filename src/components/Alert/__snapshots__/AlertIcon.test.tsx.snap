// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertIcon should render alert icon correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={16}
  bbWidth={16}
  fill="#F27A1A"
  focusable={false}
  height={16}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 16,
        "width": 16,
      },
    ]
  }
  testID="alert-icon"
  title="info"
  vbHeight={24}
  vbWidth={24}
  width={16}
  xmlns="http://www.w3.org/2000/svg"
>
  <RNSVGGroup
    fill={
      {
        "payload": 4294081050,
        "type": 0,
      }
    }
    propList={
      [
        "fill",
      ]
    }
  >
    <RNSVGPath
      clipRule={0}
      d="M12 0c6.627 0 12 5.373 12 12s-5.373 12-12 12S0 18.627 0 12 5.373 0 12 0Zm-1.5 11.5a1.5 1.5 0 0 1 3 0v5a1.5 1.5 0 0 1-3 0v-5ZM12 6a1.5 1.5 0 1 0 0 3 1.5 1.5 0 0 0 0-3Z"
      fill={
        {
          "payload": 4294081050,
          "type": 0,
        }
      }
      fillRule={0}
      propList={
        [
          "fill",
          "fillRule",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`AlertIcon should render null when icon false 1`] = `null`;
